dl/src/main/scala/com/intel/analytics/bigdl/utils/Engine.scala

```diff
@@ -322,14 +349,93 @@ object Engine {
       }
       Some(sc)
     } else {
-      physicalCoreNumber = cores
+      _onSpark = false
       None
     }
     _isInitialized = true
 
     ret
   }
 
+  private def dynamicAllocationExecutor : Option[Int] = {
+    if (System.getProperty("spark.dynamicAllocation.enabled") == "true") {
+      val maxExecutors = if (System.getProperty("spark.dynamicAllocation.maxExecutors") != null) {
+        System.getProperty("spark.dynamicAllocation.maxExecutors").toInt
+      } else {
+        1
+      }
+      val minExecutors = if (System.getProperty("spark.dynamicAllocation.minExecutors") != null) {
+        System.getProperty("spark.dynamicAllocation.minExecutors").toInt
+      } else {
+        1
+      }
+      require(maxExecutors == minExecutors, "spark.dynamicAllocation.maxExecutors and " +
+        "spark.dynamicAllocation.minExecutors must be identical in dynamic allocation")
+      Some(minExecutors)
+    } else {
+      None
+    }
+  }
+
+  private def sparkNodeAndCore : (Int, Int) = {
+    require(System.getProperty("spark.master") != null, "Can't find spark.master, do you start " +
+      "your application without spark-submit?")
+    val master = System.getProperty("spark.master")
+    if(master.toLowerCase.startsWith("local")) {
+      // Spark local mode
+      val pattern = "local\\[(\\d+)\\]".r
+      master match {
+        case pattern(n) => (1, n.toInt)
+        case _ => throw new IllegalArgumentException(s"Can't parser master $master")
+      }
+    } else if (master.toLowerCase.startsWith("spark")) {
+      // Spark standalone mode
+      val coreString = System.getProperty("spark.executor.cores")
```

Comment:
Also, it's high probability that user will submit spark without --total-executor-cores, because they want to use all available cores in standalone cluster. In such cases, you can not get these system properties?